xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceProviderManager.java

```diff
@@ -58,7 +58,16 @@ public PubSubPersistenceProvider getProvider() {
 
     private void initProvider()
     {
-        final Class clazz = PROVIDER.getValue();
+        Class clazz = PROVIDER.getValue();
```

Comment:
I probably would have left the default value in place, and do something like ...

```
clazz == ClusterManager.isClusteringEnabled()
    ? PROVIDER.getDefaultValue()
    : PROVIDER.getValue()
```
Though the ultimate end is essentially the same (enabling clustering /after/ the fact will of course cause problems!)

Reply:
I was meaning to express that this property has no (fixed) default. I thought that to be expressing intent a bit better.

Reply:
Ah I see your point. So non-cluster default = `CachingPubsubPersistenceProvider`, cluster default = `DefaultPubSubPersistenceProvider`

Perhaps `DefaultPubSubPersistenceProvider` needs a rename if it's not actually the default?

Reply:
It kind of is - `CachingPubsubPersistenceProvider` will delegate. I could've named it JDBC- I guess, but that'd conflict with potential future ones that also operate on a database.